Review Request

themohammadsa opened the pull request:
Hey everyone! I am done with the v1.0 of my ecommerce application - gortiKart. 

I have added the following features into gortiKart:

--> Filter Features based on categories and Sort Features
--> Search Results on the products page with URL change
--> Responsiveness for Mobiles, Tablets, Desktop and HD Desktops
--> Separate landing pages for different categories
--> Imported database from mongoDB
--> Redirecting button in cart for user if wishlist has pending items and cart is empty and vice versa

Pending features to add later includes: Auth, Address & User Management on the Backend

Would love your reviews, and honest feedbacks on it. Will keep updating based on it! :D

Here's the link:
https://gortikart.netlify.app/

hntejas reviewed (commented):
Hi Mohammad, 

The application looks very clean and responsive. 
Here are some of my observations and suggestions: 

Functional: 
1) Ability to adjust quantity of products in Cart Page - this could avoid duplicate addition of products also provide a better UX 
2) As a side effect removing one item is removing all items of the same name 

Code: 
1) Styles can be divided into different files, as it's difficult to debug things from a single file also to manage 
2) In a similar line, reducer function file also can be split to address the functionailty (optional - just keeping in mind the scalability)
3) Folder names can be kept consistent with same case (Context and Pages folder)

kajol-rep commented on src/components/Categorise/Mobile/CategoriseMobile.js:
```diff
@@ -0,0 +1,39 @@
+        </div>
+      </div>
+
+      {blockSortDisplay ? <Sort /> : null}
+
+      {blockFilterDisplay ? <Filter /> : null}
```
Here you can change the state variable name to something like displaySort and displayFilter, because blockSortDisplay seems to give a different meaning than what you are doing here.
Or maybe you can just change the logic to this:

```suggestion
      {blockSortDisplay ? null : <Sort />}

      {blockFilterDisplay ? null : <Filter />}

```

kajol-rep reviewed (changes requested):
Great work. App looks good.
I have suggested a fix you can take a look at it.

themohammadsa commented:
> Great work. App looks good.
> I have suggested a fix you can take a look at it.

themohammadsa commented:
> Hi Mohammad,
> 
> The application looks very clean and responsive.
> Here are some of my observations and suggestions:
> 
> Functional:
> 
> 1. Ability to adjust quantity of products in Cart Page - this could avoid duplicate addition of products also provide a better UX
> 2. As a side effect removing one item is removing all items of the same name
> 
> Code:
> 
> 1. Styles can be divided into different files, as it's difficult to debug things from a single file also to manage
> 2. In a similar line, reducer function file also can be split to address the functionailty (optional - just keeping in mind the scalability)
> 3. Folder names can be kept consistent with same case (Context and Pages folder)

Hello Tejas!
Thank you for taking your time to review the code.

I have made the necessary changes required as per your suggestions. 

Functional: 
1. I'll update this after I complete my Authentication for users.
2. I have added filters, so that this problem is avoided. 

Code:

1. The separation of styles.css, I am working on it and will update it as soon as possible. It takes a lot of time as I didn't follow this practice from the start.
2. Inference after recent discussion with you is - since there's just one reducer function being used, one file is okay 
3. I have fixed this issue.

themohammadsa commented:
> Great work. App looks good.
> I have suggested a fix you can take a look at it.

Hello Kajol!
Thank you for taking your time to review the code.

I have made the necessary changes as per your suggestions. Please do verify it.

kajol-rep reviewed (approved):
Looks better now. Please go ahead.

hntejas commented:
> > Hi Mohammad,
> > The application looks very clean and responsive.
> > Here are some of my observations and suggestions:
> > Functional:
> > 
> > 1. Ability to adjust quantity of products in Cart Page - this could avoid duplicate addition of products also provide a better UX
> > 2. As a side effect removing one item is removing all items of the same name
> > 
> > Code:
> > 
> > 1. Styles can be divided into different files, as it's difficult to debug things from a single file also to manage
> > 2. In a similar line, reducer function file also can be split to address the functionailty (optional - just keeping in mind the scalability)
> > 3. Folder names can be kept consistent with same case (Context and Pages folder)
> 
> Hello Tejas!
> Thank you for taking your time to review the code.
> 
> I have made the necessary changes required as per your suggestions.
> 
> Functional:
> 
> 1. I'll update this after I complete my Authentication for users.
> 2. I have added filters, so that this problem is avoided.
> 
> Code:
> 
> 1. The separation of styles.css, I am working on it and will update it as soon as possible. It takes a lot of time as I didn't follow this practice from the start.
> 2. Inference after recent discussion with you is - since there's just one reducer function being used, one file is okay
> 3. I have fixed this issue.

Great Mohammad, go to merge from my end
